pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -190,15 +213,48 @@ func (a *AgentClusterInstall) Generate(dependencies asset.Parents) error {
 		}
 
 		if installConfig.Config.Platform.BareMetal != nil {
+			baremetalPlatform := agentClusterInstallOnPremPlatform{}
+			bmIcOverridden := false
+
 			if len(installConfig.Config.Platform.BareMetal.APIVIPs) > 1 {
+				bmIcOverridden = true
+				baremetalPlatform.APIVIPs = installConfig.Config.Platform.BareMetal.APIVIPs
+				baremetalPlatform.IngressVIPs = installConfig.Config.Platform.BareMetal.IngressVIPs
+			}
+
+			for _, host := range agentHosts.Hosts {
+				// Override if BMC values are not the same as default
+				if host.BMC.Username != "" || host.BMC.Password != "" || host.BMC.Address != "" || host.BMC.DisableCertificateVerification {
+					bmhost := baremetal.Host{
+						Name: host.Hostname,
+						Role: host.Role,
+					}
+					if len(host.Interfaces) > 0 {
+						// Boot MAC address is stored as first interface
+						bmhost.BootMACAddress = host.Interfaces[0].MacAddress
+					} else {
+						logrus.Infof("Could not obtain baremetal BootMacAddress for %s", installConfig.Config.Platform.Name())
+					}
+					bmIcOverridden = true
+					bmhost.BMC = host.BMC
+					baremetalPlatform.Hosts = append(baremetalPlatform.Hosts, bmhost)
+
+					// Set provisioning network configuration
+					baremetalPlatform.ClusterProvisioningIP = installConfig.Config.Platform.BareMetal.ClusterProvisioningIP
+					baremetalPlatform.ProvisioningNetwork = installConfig.Config.Platform.BareMetal.ProvisioningNetwork
+					baremetalPlatform.ProvisioningNetworkInterface = installConfig.Config.Platform.BareMetal.ProvisioningNetworkInterface
+					baremetalPlatform.ProvisioningNetworkCIDR = installConfig.Config.Platform.BareMetal.ProvisioningNetworkCIDR
```

Comment:
I think some additional validation is needed here. I tested a scenario where the bmc credentials are provided, but the provisioningNetwork, provisioningNetworkInterface, and provisioningNetworkCIDR were missing or field name misspelled. The cluster installed but the BMH's state was empty where `externally provisioned` was expected. I think validations to check that provisionNetwork is set if BMH hosts are specified and vice versa should be added.

Reply:
Shouldn't that be something already addressed by the existing baremetal platform validations?  Ie, the goal here is to reuse an already existing configuration, which needs to get validated as usual

Reply:
We do have existing baremetal provisioning validations that are being used e.g. https://github.com/openshift/installer/blob/master/pkg/types/baremetal/validation/platform.go#L132, one of the unit tests I added shows these validations being hit. 
What Richard noted is that if either the ProvisioningNetwork is Disabled or the ProvisioningNetworkInterface is not set then the BMC will not be `Externally Provisioned.` Note that the ProvisioningNetworkCIDR or ProvisioningIP have default values if not set and are checked in the existing validations IF set. 

So a validation for the ProvisioningNetwork and ProvisioningNetworkInterface when BMC is set would be useful, I will add it.

Reply:
Ok, that sounds reasonable, I initially thought it was a broader indication. Thanks for clarifying it.

Reply:
Added a warning if the BMC is configured on a host but provisioning is not configured - either because the ProvisioningInterface is not set or ProvisioningNetwork is not set to `Managed`